src/QsCompiler/Compiler/CompilationLoader.cs

```diff
@@ -574,7 +574,10 @@ public CompilationLoader(SourceLoader loadSources, ReferenceLoader loadReference
             {
                 // TODO: It would be nicer to trim unused intrinsics. Currently, this is not possible due to how the old setup of the C# runtime works.
                 // With the new setup (interface-based approach for target packages), it is possible to trim ununsed intrinsics.
-                var rewriteStep = new LoadedStep(new SyntaxTreeTrimming(keepAllIntrinsics: !qirEmissionEnabled), typeof(IRewriteStep), thisDllUri);
+#pragma warning disable CS0618 // Type or member is obsolete
+                // TODO: The dependencies for rewrite steps should be declared as part of IRewriteStep interface, and we should query those here.
+                var rewriteStep = new LoadedStep(new SyntaxTreeTrimming(keepAllIntrinsics: !qirEmissionEnabled, BuiltIn.AllBuiltIns.Select(e => e.FullName)), typeof(IRewriteStep), thisDllUri);

```

Comment:
We had previously filtered out built-in attributes as well. Not sure if you want to keep those or not.

Reply:
It seems more consistent to keep them if we keep the rest with the argument of it being listed in BuiltIn.